api/src/main/java/org/openmrs/api/impl/AdministrationServiceImpl.java

```diff
@@ -102,10 +106,20 @@ public class AdministrationServiceImpl extends BaseOpenmrsService implements Adm
 	@Qualifier("implementationIdHttpClient")
 	private HttpClient implementationIdHttpClient;
 	
-	/**
-	 * Default empty constructor
-	 */
-	public AdministrationServiceImpl() {
+	private final MessageSourceService messageSourceService;
+	private final SerializationService serializationService;
+	private final ConceptService conceptService;
+	private final AdministrationService self;
+	
+	@Autowired
+	public AdministrationServiceImpl(@Lazy MessageSourceService messageSourceService,
+	                                @Lazy SerializationService serializationService,
+	                                @Lazy ConceptService conceptService,
+	                                @Lazy AdministrationService self) {
+		this.messageSourceService = messageSourceService;
+		this.serializationService = serializationService;
+		this.conceptService = conceptService;
+		this.self = self;
 	}
```

Comment:
Let's have an extra constructor without a `self` parameter that assigns `this.self = this;` so that the class can be easily mocked and is not fully dependent on Spring.